web/src/click_handlers.js

```diff
@@ -858,7 +858,7 @@ export function initialize() {
             popovers.hide_all(not_hide_tippy_instances);
         }
 
-        if (compose_state.composing()) {
+        if (compose_state.composing() && !$(e.target).parents("#compose").length) {
```

Comment:
This change could potentially cause clicks on compose box buttons or menu items (say, in the giphy picker) to fail to return focus to the compose box. We should test that carefully -- while the intent is definitely not for the `a` handler below to run in those cases, it's very possible that the developers of those features didn't bother writing the line or two code to refocus the compose box that they should have because it happened to work due to this structure.

Reply:
I did click around on several things to test this, but I can do a more thorough review that's recorded for your review. Here:


https://user-images.githubusercontent.com/5634097/234131615-49ba284e-b42a-4fde-9339-8e7747e75621.mp4

Notably exiting modals doesn't bring focus back, but it also doesn't on CZO so that isn't new.